encoding/build.js

```diff
@@ -0,0 +1,465 @@
+      required: true
+    },
+    {
+      name: 'systemLength',
+      type: 'uint',
+      required: true
+    },
+    {
+      name: 'indexersUpdated',
+      type: 'bool',
+      required: true
+    },
+    {
+      name: 'fastForwarding',
+      type: 'bool',
+      required: true
+    },
+    {
+      name: 'recoveries',
+      type: 'uint',
+      required: false
+    }
+  ]
+})
+
+autobase.register({
+  name: 'bootRecord',
+  versions: [
+    {
+      version: 0,
+      type: '@autobase/bootRecordV0'
+    },
+    {
+      version: 3,
+      type: '@autobase/bootRecordRaw'
+    }
+  ]
+})
+
+autobase.register({
+  name: 'checkpointer',
+  compact: true,
+  fields: [
+    {
+      name: 'checkpointer',
+      type: 'uint',
+      required: true
+    },
+    {
+      name: 'checkpoint',
+      type: '@autobase/indexCheckpoint',
+      required: false
+    }
+  ]
+})
+
+autobase.register({
+  name: 'checkpointerArray',
+  array: true,
+  type: '@autobase/checkpointer'
+})
+
+autobase.register({
+  name: 'checkpoint',
+  compact: false,
+  fields: [
+    {
+      name: 'system',
+      type: '@autobase/checkpointer',
+      required: false
+    },
+    {
+      name: 'encryption',
+      type: '@autobase/checkpointer',
+      required: false
+    },
+    {
+      name: 'user',
+      type: '@autobase/checkpointerArray',
+      required: false
+    }
+  ]
+})
+
+autobase.register({
+  name: 'indexer',
+  compact: true,
+  fields: [
+    {
+      name: 'signature',
+      type: 'uint',
+      required: true
+    },
+    {
+      name: 'namespace',
+      type: 'fixed32',
+      required: true
+    },
+    {
+      name: 'publicKey',
+      type: 'fixed32',
+      required: true
+    }
+  ]
+})
+
+autobase.register({
+  name: 'indexers',
+  array: true,
+  type: '@autobase/indexer'
+})
+
+autobase.register({
+  name: 'digest',
+  compact: false,
+  fields: [
+    {
+      name: 'pointer',
+      type: 'uint',
+      required: true
+    },
+    {
+      name: 'key',
+      type: 'fixed32',
+      required: false
+    }
+  ]
+})
+
+autobase.register({
+  name: 'node',
+  compact: true,
+  fields: [
+    {
+      name: 'heads',
+      type: '@autobase/clock',
+      required: true
+    },
+    {
+      name: 'batch',
+      type: 'uint',
+      required: true
+    },
+    {
+      name: 'value',
+      type: 'buffer',
+      required: true
+    }
+  ]
+})
+
+autobase.register({
+  name: 'oplogMessageV0',
+  external: 'OplogMessageV0'
+})
+
+autobase.register({
+  name: 'oplogMessageV1',
+  external: 'OplogMessageV1'
+})
+
+autobase.register({
+  name: 'oplogMessageV2',
+  compact: false,
+  fields: [
+    {
+      name: 'checkpoint',
+      type: '@autobase/checkpoint',
+      required: false
+    },
+    {
+      name: 'digest',
+      type: '@autobase/digest',
+      required: false
+    },
+    {
+      name: 'optimistic',
+      type: 'bool'
+    },
+    {
+      name: 'node',
+      type: '@autobase/node',
+      required: true
+    }
```

Comment:
lets move this to be the first part of the message since its required and compact, my guess is that in a future refactor we will inline it anyway